app/src/main/java/com/example/pix/home/fragments/ComposeFragment.java

```diff
@@ -1,42 +1,219 @@
 package com.example.pix.home.fragments;
 
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.hardware.Camera;
 import android.os.Bundle;
 
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import android.view.GestureDetector;
+import android.view.Gravity;
 import android.view.LayoutInflater;
+import android.view.MotionEvent;
+import android.view.PixelCopy;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Button;
 import android.widget.FrameLayout;
-import com.google.ar.core.Config;
-import com.google.ar.core.Session;
-import com.google.ar.sceneform.ux.ArFragment;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.PopupWindow;
+import android.widget.SearchView;
+import android.widget.Toast;
+
+import com.example.pix.R;
+import com.example.pix.chat.ChatActivity;
+import com.example.pix.chat.ChatFragment;
+import com.example.pix.home.activities.HomeActivity;
+import com.example.pix.home.adapters.SearchAdapter;
+import com.example.pix.home.utils.CameraPreview;
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ComposeFragment extends Fragment {
 
-import java.util.EnumSet;
-import java.util.Set;
+    ChatActivity mActivity;
```

Comment:
Please add `private` or `public` or `protected` for this member and other members in all other classes.